source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2696 @@
+  `GridFS <../gridfs/gridfs-spec.rst>`__ spec.
+
+  The structure of this object is as follows:
+
+  - ``id``: Required string. Unique name for this entity. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g. ``id: &bucket0 bucket0``).
+
+  - ``database``: Required string. Database entity from which this bucket will
+    be created. The YAML file SHOULD use an `alias node`_ for a database
+    entity's ``id`` field (e.g. ``database: *database0``).
+
+  - ``bucketOptions``: Optional object. Additional options used to construct
+    the bucket object. Supported options are defined in the
+    `GridFS <../source/gridfs/gridfs-spec.rst#configurable-gridfsbucket-class>`__
+    specification. The ``readConcern``, ``readPreference``, and ``writeConcern``
+    options use the same structure as defined in `Common Options`_.
+
+.. _entity_stream:
+
+- ``stream``: Optional object. Defines a stream, as defined in the
+  `GridFS <../gridfs/gridfs-spec.rst>`__ spec. Test runners MUST ensure that
+  stream is both readable *and* writable.
+
+  This entity is primarily used with `uploadFromStream`_ and
+  `uploadFromStreamWithId`_ operations for `bucket`_ entities.
+
+  The structure of this object is as follows:
+
+  - ``id``: Required string. Unique name for this entity. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g. ``id: &stream0 stream0``).
+
+  - ``hexBytes``: Required string. The string MUST contain an even number of
+    hexademical characters (case-insensitive) and MAY be empty. The test runner
+    MUST raise an error if the string is malformed. The test runner MUST convert
+    the string to a byte sequence denoting the stream's readable data (if any).
+    For example, "12ab" would denote a stream with two bytes: "0x12" and "0xab".
+
+
+collectionData
+~~~~~~~~~~~~~~
+
+List of documents that should correspond to the contents of a collection. This
+structure is used by both `initialData`_ and `test.outcome <test_outcome_>`_,
+which insert and read documents, respectively.
+
+The structure of this object is as follows:
+
+- ``collectionName``: Required string. See `commonOptions_collectionName`_.
+
+- ``databaseName``: Required string. See `commonOptions_databaseName`_.
+
+- ``documents``: Required array of objects. List of documents corresponding to
+  the contents of the collection. This list may be empty.
+
+
+test
+~~~~
+
+Test case consisting of a sequence of operations to be executed.
+
+The structure of this object is as follows:
+
+- ``description``: Required string. The name of the test.
+
+  This SHOULD describe the purpose of this test (e.g. "insertOne is retried").
+
+.. _test_runOnRequirements:
+
+- ``runOnRequirements``: Optional array of on or more `runOnRequirement`_
+  objects. List of server version and/or topology requirements for which this
+  test can be run. If specified, these requirements are evaluated independently
```

Comment:
I think "independent" here is confusing, I suggest removing it leaving only "these requirements are evaluated in addition to any top-level requirements".

Reply:
My worry with "in addition" is that test runners might simply combine the top-level `runOnRequirements` with the test-level requirements. Both arrays need to be evaluated independently. If a top-level requirement is met but no test-level requirements are met, the test must be skipped, and vice versa.

Reply:
I see, I cannot think of an improvement to the current wording therefore I'll resolve this thread.